src/backend/distributed/utils/metadata_cache.c

```diff
@@ -933,6 +934,52 @@ HasUninitializedShardInterval(ShardInterval **sortedShardIntervalArray, int shar
 
 
 /*
+ * HasOverlappingShardInterval determines whether the given list of sorted
+ * shards has overlapping ranges.
+ */
+static bool
+HasOverlappingShardInterval(ShardInterval **shardIntervalArray,
```

Comment:
We'll need to fix a lot more code if we stop assuming distinct ranges, so this seems pretty arbitrary